New print styles and generator to add to build

jathayde opened the pull request:
<img width="958" alt="Screenshot 2024-03-31 at 8 43 35 PM" src="https://github.com/rails/rails/assets/2223/995815e6-1d34-4550-b49c-c7a78e213cc1">

### Motivation / Background

This updates print CSS to be more inline with the overall style. The sidebar is inlined (same as mobile), and content is 100%. Previous code in the launch from boilerplate has been struck or integrated better.

* Links show in text after the original link text
* Border line around code and copy button are removed
* Print is now generated from SCSS to CSS via the generator that builds the other CSS files 

### Additional information

Some browsers will override backgrounds on print

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

jathayde committed:
New print styles and generator to add to build

jathayde committed:
Reducing font size to 10pt from 12 pt

jathayde committed:
Fixing background prints, removing errant border

carlosantoniodasilva commented on guides/assets/stylesrc/print.scss:
```diff
@@ -0,0 +1,180 @@
+    background: transparent;
+    font-weight: bold;
+    text-decoration: underline;
+    }
```
Also, maybe move these link definitions up with the others that are near the beginning, just to keep them close together. (lines 21-24)

carlosantoniodasilva reviewed (commented):
@jathayde nice! it's great to have some good print styles for these.

I had a few bits of feedback below, most importantly the previous print.css source file that we need to remove.

I was also wondering if we had any control over the page margin, it seems like the top could get some extra space, see the screenshots below.

<details><summary>Margin Samples</summary>
<p>

![Screenshot 2024-04-09 at 11 00 23](https://github.com/rails/rails/assets/26328/311332a2-5b3d-4fa2-b68e-261401c5cc46)
![Screenshot 2024-04-09 at 11 00 34](https://github.com/rails/rails/assets/26328/b961a24d-bc30-4393-8b64-5e4b63902f08)
![Screenshot 2024-04-09 at 11 00 30](https://github.com/rails/rails/assets/26328/b1e812c5-2e28-4ce6-8613-bff518dd4d83)

</p>
</details> 

Totally cool to leave it if that's not something we have much control over, it was just something I noticed on a quick scan through the printed PDF.

jathayde committed:
Setting the `@page` element and defining the margins explicitly

jathayde committed:
Explicitly defining `pre` font size for code blocks

jathayde committed:
Removing generated print.css since it generates in build

jathayde committed:
Indent fix, moving anchor definitions together.

jathayde commented:
Setting the `@page` element and defining the margins explicitly (5f06956b):

<img width="641" alt="Screenshot 2024-04-16 at 9 52 46 AM" src="https://github.com/rails/rails/assets/2223/fd81a47f-12df-4442-a13f-22425898e4fe">

Explicitly defining `pre` font size for code blocks to make it feel more inline with the body (9pt in IBM Plex Mono vs 10pt in Inter body) (61132c72)

<img width="261" alt="Screenshot 2024-04-16 at 9 54 56 AM" src="https://github.com/rails/rails/assets/2223/f2aa9743-94c4-4d2e-810c-6fe7b618f033">

Removed static print.css in e5e7beca

Fixed anchor considerations in b69c5c4a

carlosantoniodasilva reviewed (commented):
Thanks for the updates @jathayde! I think the print styles are looking pretty great now, there's just one thing that is not sitting super right: the inner code pieces still seem to have bigger font-size, causing them to standout a little too much.

<details><summary>Here's a few examples</summary>
<p>

![Screenshot 2024-04-17 at 14 09 07](https://github.com/rails/rails/assets/26328/3f2ca9d3-6f18-487d-8e8e-b602b613a1aa)

![Screenshot 2024-04-17 at 14 10 23](https://github.com/rails/rails/assets/26328/494400de-75b6-4b52-88fa-2c72f634e3ae)

![Screenshot 2024-04-17 at 14 10 40](https://github.com/rails/rails/assets/26328/ba73e762-9c2a-47d2-9ea8-beeb4afd3aac)

![Screenshot 2024-04-17 at 14 10 44](https://github.com/rails/rails/assets/26328/19c8a20f-fdb4-4989-8aca-641d3ae48d69)

</p>
</details> 

Even some of the code blocks get parts of the code with different font-size, I'm not sure exactly which element is that, but we might need to target it somehow to adjust the font-size accordingly, otherwise it feels there's too much variation between normal text, inner code highlight, and some of the elements in code blocks. I hope the screenshots can show that, but let me know if you need more info. Thanks!

jathayde commented:
Ah this is a cascade failure. There's a separate call for inline stuff that's throwing this. Let me get that change in.

jathayde committed:
fixing the oversizes inline code font

jathayde commented:
<img width="863" alt="Screenshot 2024-04-18 at 8 12 37 AM" src="https://github.com/rails/rails/assets/2223/393c3b25-156a-49d2-9ae3-020e458b5f65">
<img width="823" alt="Screenshot 2024-04-18 at 8 13 59 AM" src="https://github.com/rails/rails/assets/2223/504911d7-bce5-4e3f-b4cd-0322033f584b">

@carlosantoniodasilva I've moved the `pre` and `code` calls globally in print.css to 9.5, and then also they exist under the `.code` interstitial call as well (to override the original css calls)

carlosantoniodasilva commented:
@jathayde thanks! The updates are looking great.

I noticed one more minor thing on a quick final pass, that I wanted to hear your thoughts whether it's something worth looking into / potentially changing.

Code inline elements that are also links appear to be super bold, where they also standout a bit too much, see the screenshots for reference:

<details><summary>Details</summary>
<p>

![Screenshot 2024-04-18 at 10 13 47](https://github.com/rails/rails/assets/26328/21c1b5bc-1870-4698-b0e7-e4063f4c47d1)

![Screenshot 2024-04-18 at 10 14 17](https://github.com/rails/rails/assets/26328/b8567991-110a-45a0-a2f3-07d2d6851e43)

![Screenshot 2024-04-18 at 10 14 24](https://github.com/rails/rails/assets/26328/c571908e-4c46-43be-80bf-e400134a1ef4)

</p>
</details> 

If you think it makes sense (and it's possible) to tweak that a bit, maybe decrease the bold some, then let's do it, other than that this LGTM. Thanks a bunch!

jathayde committed:
Fixing bold in anchors for code/pre

carlosantoniodasilva commented:
Ah I should've mentioned I was testing on Safari, but here's a Chrome vs Safari comparison for reference (before the last change):

<details><summary>Chrome vs Safari on Mac (before)</summary>
<p>

![Screenshot 2024-04-18 at 13 23 32](https://github.com/rails/rails/assets/26328/d3ee2828-c69b-4325-8f21-0909c378811d)

</p>
</details> 

Chrome does look nicer than Safari with bold there.

Here's the vs after the latest change:

<details><summary>Chrome vs Safari on Mac (after)</summary>
<p>

![Screenshot 2024-04-18 at 13 24 54](https://github.com/rails/rails/assets/26328/38201545-5844-416e-9228-e00f041a2bc4)

</p>
</details> 

The updated version looks great to me! Thanks.
